easybuild/easyblocks/a/adios.py

```diff
@@ -0,0 +1,93 @@
+##
+# Copyright 2020-2020 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing ADIOS, implemented as an easyblock
+
+@author: Ake Sandgren (HPC2N, Umea University)
+"""
+import os
+
+from easybuild.easyblocks.generic.cmakepythonpackage import CMakePythonPackage
+from easybuild.tools.environment import setvar
+from easybuild.tools.filetools import change_dir
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+
+
+class EB_ADIOS(CMakePythonPackage):
+    """Support for building/installing ADIOS."""
+
+    def install_step(self):
+        """Custom install step for ADIOS."""
+
+        super(EB_ADIOS, self).install_step()
+
+        # Create numpy wrappers
+        if get_software_root('SciPy-bundle'):
+            newpath = os.pathsep.join([os.getenv('PATH', ''), os.path.join(self.installdir, 'bin')])
+            setvar('PATH', newpath)
+            change_dir(os.path.join(self.cfg['start_dir'], 'wrappers', 'numpy'))
+
+            setup_list = ['']
```

Comment:
any reason why it's a list? You currently only ever push one element in it? So a string will suffice?

Reply:
The initial list contains an empty string, then if MPI is enabled I push '_mpi' to it.
Check the usage and it may become obvious :-)

Reply:
Still waiting for the user to come back with feedback on this...

Reply:
user feedback is that ADIOS works.
So,  @wpoely86, do you still have an issue with setup_list?

Reply:
:+1: